synapse/config/sso.py

```diff
@@ -74,6 +74,8 @@ def read_config(self, config, **kwargs):
 
         self.sso_client_whitelist = sso_config.get("client_whitelist") or []
 
+        self.sso_update_display_name = sso_config.get("update_display_name") or False
+
```

Comment:
While having `False` here does preserve existing behaviour, I almost feel like this option should be enabled by default, as I imagine it's expected for user details on SSO-enabled systems to update automatically.

Do you have any thoughts @clokep?

Reply:
It does seem like people would want this automatically, we should probably just make sure we document it. 🤷 

It probably depends on if people are using corporate-type SSO vs. social login sort of SSO though...

Reply:
> It does seem like people would want this automatically, we should probably just make sure we document it. :shrug:

This sounds reasonable.

> It probably depends on if people are using corporate-type SSO vs. social login sort of SSO though...

I think the majority of people would expect things to sync by default. I can also see the case where I might change my social media handle frequently, but not want my displayname in all rooms to update every time. So the option is handy.